jzp-code/boot-demo/src/main/java/com/tutorial/boot_demo/service/StudentServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.tutorial.boot_demo.service;
+
+import com.fasterxml.jackson.databind.util.StdConverter;
+import com.tutorial.boot_demo.converter.StudentConverter;
+import com.tutorial.boot_demo.dao.Student;
+import com.tutorial.boot_demo.dao.StudentRepository;
+import com.tutorial.boot_demo.dto.StudentDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+
+@Service
+public class StudentServiceImpl implements StudentService{
+
+    @Autowired
+    private StudentRepository studentRepository;
+
+    @Override
+    public StudentDTO getStudentById(long id) {
+        Student student = studentRepository.findById(id).orElseThrow(RuntimeException::new);
+        return StudentConverter.convertStudent(student);
+
+    }
+
+    @Override
+    public Long addNewStudent(StudentDTO studentDTO) {
+        List<Student> studentList = studentRepository.findByEmail(studentDTO.getEmail());
+        if (!CollectionUtils.isEmpty(studentList)) {
+            throw new IllegalStateException("email:" + studentDTO.getEmail() + " has been taken");
+        }
+        Student student = studentRepository.save(StudentConverter.convertStudent(studentDTO));
+        return student.getId();
+    }
+
+    @Override
+    public void deleteStudentById(long id) {

```

Comment:
need a `@Transactional` here?

Reply:
> need a `@Transactional` here?
yes, the `@Transactional` will make sure SQL sentense run in a single transaction to ensure data consistency and proper transaction management, sorry for ignoring it.